### PR TITLE
Permit custom views in notify popups

### PR DESCRIPTION
**Pull request by chris-steele**
issue https://github.com/adaptlearning/adapt_framework/issues/1840

Useful and particularly so to https://github.com/adaptlearning/adapt-contrib-hotgraphic/pull/144

### CODE REVIEW

**Comment by oliverfoster** [comment]
Could you turn this into accepting a view rather than a template?

Allowing a template-override-only means that the view behaviour associated with the template must be external to the notify and must seek the DOM node after render of the notify. If the view is instantiated and handed off to the notify it makes for a much cleaner API.
I would prefer if this behaved more like the drawer; in that it should accept child views. I would see this as a leapfrog step, scrapping the transition to bespoke templates and allowing bespoke, atomized behaviour, templates and styling.

**Comment by chris-steele** [comment]
@oliverfoster Notify now accepts a custom view. Let me know if you want to see a working demo with the latest hotgraphic.

File: src/core/js/views/notifyView.js [hunk 1]
```diff
@@ -60,13 +66,19 @@ define([
         },
 
         render: function() {
-            var data = this.model.toJSON();
-            var template = Handlebars.templates['notify'];
+            var data, html;
```
Comment (lc-thomasberger) [inline src/core/js/views/notifyView.js]: Out Style-Guide recommends to [declare one variable per var statement](https://github.com/adaptlearning/documentation/blob/master/01_cross_workstream/style_guide.md#declare-one-variable-per-var-statement).

File: src/core/js/views/notifyView.js [hunk 2]
```diff
@@ -60,13 +66,19 @@ define([
...
+
+            if (this.customView) {
+                html = this.customView.$el;
```
Comment (lc-thomasberger) [inline src/core/js/views/notifyView.js]: `this.customView.el` rather then `this.customView.$el` as it is passed into `jQuery.html()` ?

**Comment by oliverfoster** [comment]
Can we close this and use subviews instead of replacement views?
https://github.com/adaptlearning/adapt_framework/pull/1893

**Pull request closed without merging**